src/app/child-dev-project/notes/note-details/note-details.component.ts

```diff
@@ -1,22 +1,35 @@
-import { Component, Input, ViewChild } from "@angular/core";
-import { InteractionTypes } from "../interaction-types.enum";
+import { Component, Input, OnInit, ViewChild } from "@angular/core";
 import { Note } from "../model/note";
 import { ShowsEntity } from "../../../core/form-dialog/shows-entity.interface";
+import { ConfigService } from "../../../core/config/config.service";
+import { NoteConfig } from "./note-config.interface";
 
 @Component({
   selector: "app-note-details",
   templateUrl: "./note-details.component.html",
   styleUrls: ["./note-details.component.scss"],
 })
-export class NoteDetailsComponent implements ShowsEntity {
+export class NoteDetailsComponent implements ShowsEntity, OnInit {
   @Input() entity: Note;
   @ViewChild("dialogForm", { static: true }) formDialogWrapper;
 
-  smallScreen: boolean;
+  /** name of config array in the config json file */
+  private readonly CONFIG_ID = "notes";
 
-  interactionTypes = Object.values(InteractionTypes);
+  interactionTypes: {
+    [key: string]: {
```

Comment:
instead of re-defining the interactionType here again, maybe create an interface containing these 3 properties